### ISSUE TITLE
Style/IfUnlessModifier ignores long lines even if Metrics/LineLength is disabled

### ISSUE THREAD

**Issue by revolter** [rubocop-hq/rubocop#5538]
Enabling `Style/IfUnlessModifier` doesn't report problems with long lines, even if `Metrics/LineLength` is disabled so the line length shouldn't count:

```ruby
unless is_debug || is_beta
    sigh(app_identifier: bundle_id, output_path: File.expand_path('~/Library/MobileDevice/Provisioning Profiles'))
end
```

--------

## Expected behavior

Report a problem with the specified code.

## Actual behavior

Rubocop didn't report any problem.

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

Create this `.rubocop.yml` file:

```yaml
Metrics/LineLength:
  Enabled: false

Style/IfUnlessModifier:
  Enabled: true
```

and run `rubocop` against the provided code snippet.

## RuboCop version

```
0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.3 universal.x86_64-darwin17)
```

## Workaround

As a workaround, using this:

```yaml
Metrics/LineLength:
  Enabled: false
  Max: 9000
```

makes it report a problem, because the combined lines length isn't larger than `9000`.

**Comment by dominicsayers** [comment]
This is additionally a problem with `--auto-gen-config` because if you have `Metrics/LineLength` disabled then `--auto-gen-config` does *not* register offences for long lines, *but* then `--auto-gen-config` appends the following to the `.rubocop_todo.yml` file:

```yaml
# Configuration parameters: AllowHeredoc, AllowURI, URISchemes, IgnoreCopDirectives, IgnoredPatterns.
# URISchemes: http, https
Metrics/LineLength:
  Max: 2901
```

...so subsequent runs of `rubocop` *will* now report the offences as described by @revolter.

**Comment by dominicsayers** [comment]
I'm not sure *why* `--auto-gen-config` appends those lines to `.rubocop-todo.yml` - if that's a bug in `--auto-gen-config` I'm happy to report it separately.

**Comment by dominicsayers** [comment]
Surely [this method](https://github.com/bbatsov/rubocop/blob/150be51ad92c6c9998ab60a61a57ca87dd5e0b19/lib/rubocop/cop/mixin/statement_modifier.rb#L49) should take account of whether `Metrics/LineLength` is enabled or not?

**Comment by dominicsayers** [comment]
[This cop](https://github.com/bbatsov/rubocop/blob/aecd5eadfdcb57be213729d2e98fba7a3c02de6c/lib/rubocop/cop/style/conditional_assignment.rb#L387) seems to do it correctly, checking that `Metrics/LineLength` is enabled before looking at the `Max` setting.

**Comment by revolter** [comment]
So maybe the same thing should be added somewhere here:

https://github.com/bbatsov/rubocop/blob/031b29b495ad435dc69172fb7603042894faae53/lib/rubocop/cop/style/if_unless_modifier.rb#L33-L38
